Layouts Fixes and Normalizations

devgoncalo opened the pull request:
**Why merging this Pull Request?**
This PR fixes some minor layouts problems improving the user experience.

**What are the changes?**

1. Save SVG icon change.
- `Why?` The Save PNG and Save SVG options previously had the same icon, which affected user readability and experience. This change adds more context to the action, maybe clarifying more what the user is doing.
<img width="250" alt="after-icon" src="https://github.com/raycast/ray-so/assets/68572514/eb9552b1-d28c-4dd1-9fba-1337bfe4f370">
<img width="250" alt="before-icon" src="https://github.com/raycast/ray-so/assets/68572514/5ba650be-31a8-43f4-9694-7d33d1536c01">
<br />
<br />

2. Export Image dropdown menu width normalization (Small Devices).
- `Why?` Before this normalization, when using the dropdown menu on small devices, the menu did not respect the button width and appeared misaligned.

<img width="250" alt="before-dropdown-menu-mobile" src="https://github.com/raycast/ray-so/assets/68572514/18590a65-2726-4cd8-845f-685c20df14c3">
<img width="250" alt="after-dropdown-menu-mobile" src="https://github.com/raycast/ray-so/assets/68572514/541e99e7-0916-4e3d-b5b6-c104081b4f1c">
<br />
<br />

3. Export Image dropdown menu width normalization (Medium and Large Devices).
- `Why?` Following the same observation made on small devices, we decided to apply the normalization to medium and large devices as well.

<img width="250" alt="before-dropdown-menu-pc" src="https://github.com/raycast/ray-so/assets/68572514/72c099ec-3dac-4f53-bb03-fae83c13c626">
<img width="250" alt="after-dropdown-menu-pc" src="https://github.com/raycast/ray-so/assets/68572514/deb4b906-2edf-4d5a-8d37-f27feef8085e">
<br />
<br />

4. Code Languages select content behavior fix and normalization.
- `Why?` Previously, the Code Languages select content menu occupied the full height of the page starting from the trigger, hiding the top bar buttons and appearing misaligned and a little bit off. Now, the menu is aligned with the trigger and limited to a 550px height, resulting in a seamless behavior and layout.

<img width="250" alt="before-languages" src="https://github.com/raycast/ray-so/assets/68572514/38c93828-ac05-47ce-abc7-fe79527ebec8">
<img width="250" alt="after-languages" src="https://github.com/raycast/ray-so/assets/68572514/a4b03e14-d8b8-4eb3-a9a8-5b1803b9db31">
<br />
<br />

5. Themes select content normalization.
- `Why?` Since the Code Languages select content menu was normalized, this change was necessary. It prevents the select menu from exceeding the 550px height mark in the future. The width was also slightly increased with this change.

<img width="250" alt="before-themes" src="https://github.com/raycast/ray-so/assets/68572514/feebe69d-ba79-4397-94a4-5edeeecc946d">
<img width="250" alt="after-themes" src="https://github.com/raycast/ray-so/assets/68572514/84f27cbe-923b-4c19-aec2-9a9484abc093">
<br />
<br />

6. A branding touch in the Navigation Menu.
- `Why?` Therefore, this is not a bug or an important fix/change. I think this little hover change on Made by Raycast, making the color match the Raycast brand, adds a nice touch to the menu.

<img width="200" alt="before-made-by-raycast" src="https://github.com/raycast/ray-so/assets/68572514/05ff4381-e2f9-458e-9254-d8c7e34f902a">
<img width="200" alt="after-made-by-raycast" src="https://github.com/raycast/ray-so/assets/68572514/252ac3e8-6c35-4979-858c-92d785894a85">
<br />
<br />

7. Removal of some unused imports.
- `Why?` While making these layout changes, I encountered several unused imports and decided to clean them up for future contributions and code maintainability. Removing unused imports is beneficial because it reduces clutter, improves code readability, and can potentially enhance performance by minimizing unnecessary dependencies.

samuelkraft commented:
Hey @devgoncalo, thanks for the contribution!
The dropdown content normalization is nice, but I just added keyboard shortcut hints to all dropdowns as well as changing the navigation menu a bit making this outdated. The current select appearance are by design full height and item-aligned. 

The SVG icon doesn't feel super recognizable and has the wrong stroke thickness, maybe try a bezier curve icon or similar?

devgoncalo commented:
After reviewing the recent changes, I have devised a solution and made a few additional modifications, outlined as follows:

1. I ensured that the "Export Image" full text is displayed on all devices while keeping the shortcut (KBD) hidden on small devices, which are predominantly mobile therefore the shortcuts may not be used at all. This maintains the dropdown menu normalization on small devices without compromising user space and preserves shortcut information on larger devices.
<img width="200" alt="before-code-dropdown-menu" src="https://github.com/raycast/ray-so/assets/68572514/c45c97f6-b1b6-4123-b1be-f51980fbffa5">
<img width="200" alt="after-code-dropdown-menu" src="https://github.com/raycast/ray-so/assets/68572514/bf81bfbc-c530-4b7a-8339-fa11b141acd4">
<br />
<br />
<img width="250" alt="before-code-dropdown-menu" src="https://github.com/raycast/ray-so/assets/68572514/3afe7884-4118-42b1-abd7-4cd31024d1cf">
<img width="250" alt="after-code-dropdown-menu" src="https://github.com/raycast/ray-so/assets/68572514/9b518325-0ae7-445d-b546-cf035d341a1b">
<br />
<br />

2. Since the original SVG icon was not recognizable, I replaced the SVG icon with a new path-based icon resembling the contents of an SVG that are paths. The stroke is maintained at 2 to ensure the icon's clarity and consistency; a thinner stroke of would 1.5 compromises the application's icon integrity.
<img width="250" alt="svg-icon-before" src="https://github.com/raycast/ray-so/assets/68572514/8cf44c6f-9d0f-436c-a76c-e80f9cfca47e">
<img width="250" alt="svg-icon-after" src="https://github.com/raycast/ray-so/assets/68572514/90d2b849-54fc-4b38-8bfa-32339d13e941">
<br />

3. On mobile devices, the onShare function was malfunctioning not allowing the users to share their icons. And for some reason the dropdown menu was hidden on small devices preventing users from downloading, copying, or sharing icons. I removed this function and retained the dropdown menu, providing users with the same functionality across all device sizes.

<img width="250" alt="code-button-text-before" src="https://github.com/raycast/ray-so/assets/68572514/17822b5d-5dce-47eb-b8f6-9edaefbc9056">
<img width="250" alt="code-button-text-after"src="https://github.com/raycast/ray-so/assets/68572514/d44a566a-dc57-4093-8f6a-ed39935e48f7">
<br />
<br />
<img width="250" alt="icon-dropdown-menu-before" src="https://github.com/raycast/ray-so/assets/68572514/75d113ac-4621-4045-a724-dd5c338344ed">
<img width="250" alt="icon-dropdown-menu-after" src="https://github.com/raycast/ray-so/assets/68572514/1e08d48c-26a3-429f-84f5-cf881f975075">
<br />

4. Lastly, I believe that the full-height design for select elements such as the code themes and code languages is not optimal, as it disrupts user flow and appears cramped. I have retained the changes but I am open to your feedback, and if there is a reason to know it!
